Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -2230,7 +2232,29 @@ namespace AZ
                             propertyIndex = material->FindPropertyIndex(s_emissive_intensity_Name);
                             if (propertyIndex.IsValid())
                             {
-                                subMesh.m_emissiveColor *= material->GetPropertyValue<float>(propertyIndex);
+                                auto unitPropertyIndex = material->FindPropertyIndex(s_emissive_unit_Name);
+                                AZ_Assert(
```

Comment:
Maybe this should be a warning so it's visible when not in debug?
```suggestion
                                AZ_WarningOnce(
```

Reply:
In addition to changing this to a warning, the rest of the logic should be wrapped in an if block. No reason to execute any of the other logic if the index is invalid. That's consistent with checks involving other material property indexes.